ethcontract/src/transaction/typed_gas_price.rs

```diff
@@ -0,0 +1,229 @@
+//! Implementation of typed gas price estimation.
+
+use crate::errors::ExecutionError;
+use crate::GasPrice;
```

Comment:
Personally, I'd be OK with getting rid of the `GasPrice` thing that we added a while back. The intention was to "scale the web3 gas estaimte by a factor", but it ended up never really being used or that useful. I imagine removing it would simplify things.

I'm not sure what the others things about it.

Reply:
Yes, I believe only GasPrice::Value variant is used in the code, not sure if other variants have any purpose, expecially since we can call node.gas_price() from the gas price estimator.
Removing it would definitely make code much cleaner.

Reply:
Great! Lets purge it with fire then! :stuck_out_tongue: